front/app/api/project_library_countries/keys.ts

```diff
@@ -1,9 +1,6 @@
-import { QueryKeys } from 'utils/cl-react-query/types';
-
-const baseKey = { type: 'project_library_tenant_countries' };
-
 const projectLibraryCountriesKeys = {
-  all: () => [baseKey],
-} satisfies QueryKeys;
+  all: (locale?: string) =>
+    ['project_library_tenant_countries', ...(locale ? [locale] : [])] as const,
+};
```

Comment:
This looks a bit off to me - I think we should be using something closer to this format, rather than the "all":
```
  list: (params: IQueryParameters) => [
    { ...baseKey, operation: 'list', parameters: params },
  ],
```

I'll have a deeper look later today though :)

Reply:
@amanda-anderson I have pushed some changes that align this more with the pattern in many other examples in our codebase (e.g. for adminPublications, [here](https://github.com/CitizenLabDotCo/citizenlab/blob/master/front/app/api/admin_publications/keys.ts)).

This involves defining a new type for the locale param.

This also seems to work fine, but seems a very verbose solution for such a simple param. E.g; I now seem to be using `lists` for what is essentially a single string param 🤷

Reply:
Nice! Looks good to me :D I know it's a bit verbose, but I think it's also better for caching purposes. The way it is now, It should also cache each locale-specific list of the countries :thinking: 

Since locale is a query param, it also makes sense to me that the "all" would be for fetching all project_library_tenant_countries (without a locale param), and then if you want a specific locale version of that it would be a list with a locale query param :+1:

Reply:
Ahh, that makes sense! Thanks for explaining!! 🧑‍🎓

Reply:
> Since locale is a query param, it also makes sense to me that the "all" would be for fetching all project_library_tenant_countries (without a locale param), and then if you want a specific locale version of that it would be a list with a locale query param

Yeah, this endpoint is unusual, I guess. If no locale provided it will send the list with English names. If locale param provided, the names will be translated, but the list is otherwise the same. It's not a filter, and it doesn't reduce the n of items returned, rather it transforms the items returned.

My hunch is this doesn't matter, and it probably works just fine as it is now.